apps/hash-ai-worker-ts/src/activities/shared/get-llm-response.ts

```diff
@@ -666,15 +689,170 @@ const getOpenAiResponse = async (
  * `model` provided in the parameters.
  */
 export const getLlmResponse = async <T extends LlmParams>(
-  params: T,
+  params: T & {
+    /**
+     * Required for tracking usage on a per-user basis.
+     *
+     * @todo: consider abstracting this in a wrapper method, or via
+     * generic params (via a `logUsage` method).
+     */
+    userAccountId: AccountId;
+    graphApiClient: GraphApi;
+    linkUsageRecordToEntities: {
+      linkEntityTypeId: [
+        typeof systemLinkEntityTypes.incurredIn.linkEntityTypeId,
+        typeof systemLinkEntityTypes.created.linkEntityTypeId,
+        typeof systemLinkEntityTypes.updated.linkEntityTypeId,
+      ][number];
+      entityId: EntityId;
```

Comment:
Need some way of tracking the `webId` to link to this usage, which could be passed in here... I think best to discuss sync

Reply:
Following our conversation, I've added the `webId` parameter for the `runFlow` temporal workflow, so that usage records/other entities in the flow are created in that web.

Relevant commits: https://github.com/hashintel/hash/pull/4379/commits/b08d4e839ff8a64e2e2f5cdb8f69a2a42993bfec, https://github.com/hashintel/hash/pull/4379/commits/342dc120f4c05755a91e0456e204412a52ffc25d